modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/portlet/action/EditCommerceOrderMVCRenderCommand.java

```diff
@@ -63,13 +63,16 @@ public String render(
 					_commerceOrderHttpHelper.getCurrentCommerceOrder(
 						_portal.getHttpServletRequest(renderRequest));
 
-				if ((currentCommerceOrder == null) ||
+				if ((currentCommerceOrder != null) &&
 					(commerceOrder.getCommerceOrderId() !=
 						currentCommerceOrder.getCommerceOrderId())) {
 
 					_commerceOrderHttpHelper.setCurrentCommerceOrder(
 						_portal.getHttpServletRequest(renderRequest),
 						commerceOrder);
+					_commerceOrderHttpHelper.setCurrentCommerceOrder(
```

Comment:
I don't get this one, you are overriding the order set 2 lines before.
You are always doing that, with no specif check (if something or whatever)

Reply:
@luca-pellizzon , I was not sure about this one. I will fix accordingly.